### ISSUE TITLE
test.py error: unregonized arguments

### ISSUE THREAD

**Issue by jonahduncan** [yhlleo/DeepSegmentor#13]
You've put together a fantastic resource here, @yhlleo. Thank you for that.

I'm attempting to perform a test run of RoadNet using the pretrained model you've provided on your dataset, but keep encountering the same error message. When I run `sh ./scripts/test_roadnet.sh 0` I get the following print out:
```
usage: test.py [-h] --dataroot DATAROOT [--name NAME] [--gpu_ids GPU_IDS]
               [--checkpoints_dir CHECKPOINTS_DIR] [--model MODEL]
               [--input_nc INPUT_NC] [--output_nc OUTPUT_NC] [--ngf NGF]
               [--ndf NDF] [--netD NETD] [--netG NETG]
               [--n_layers_D N_LAYERS_D] [--norm NORM] [--init_type INIT_TYPE]
               [--init_gain INIT_GAIN] [--no_dropout]
               [--dataset_mode DATASET_MODE] [--direction DIRECTION]
               [--serial_batches] [--num_threads NUM_THREADS]
               [--batch_size BATCH_SIZE] [--load_size LOAD_SIZE]
               [--crop_size CROP_SIZE] [--max_dataset_size MAX_DATASET_SIZE]
               [--preprocess PREPROCESS] [--no_flip NO_FLIP]
               [--display_winsize DISPLAY_WINSIZE] [--epoch EPOCH]
               [--load_iter LOAD_ITER] [--verbose] [--suffix SUFFIX]
               [--load_width LOAD_WIDTH] [--load_height LOAD_HEIGHT]
               [--use_augment] [--display_sides DISPLAY_SIDES]
               [--num_classes NUM_CLASSES] [--use_selu USE_SELU]
               [--loss_mode LOSS_MODE] [--ntest NTEST]
               [--results_dir RESULTS_DIR] [--aspect_ratio ASPECT_RATIO]
               [--phase PHASE] [--eval] [--num_test NUM_TEST]
test.py: error: unrecognized arguments: 0
```

As I'm relatively inexperienced when it comes to a project of this scale, and I was unable to figure out why the message was occurring myself, I thought I'd get in touch. I'm wondering if it could be an issue with my set up as I'm working on Windows and have been unable to use conda, working from my own virtual environment instead.

**Comment by yhlleo** [comment]
Hi, did you change anything in `test_roadnet.sh`? It seems that there was an unrecognized arguments `0`, which I used for setting the gpu id. @jonahduncan

**Comment by jonahduncan** [comment]
The only modification that I made was to change python3 to python as that's what points to the python version for my set-up.

**Comment by yhlleo** [comment]
Do you solve the problem? I close this issue first, if you still have problem, I can reopen it later.

**Comment by jonahduncan** [comment]
@yhlleo I have not yet been able to solve this issue

**Comment by yhlleo** [comment]
If you revised `GPU_IDS=$1` to something like `GPU_IDS=0`, you don't need to call the script in this way: `sh ./scripts/test_roadnet.sh 0`. In this case, please call `sh ./scripts/test_roadnet.sh` directly.

**Comment by jonahduncan** [comment]
> If you revised `GPU_IDS=$1` to something like `GPU_IDS=0`, you don't need to call the script in this way: `sh ./scripts/test_roadnet.sh 0`. In this case, please call `sh ./scripts/test_roadnet.sh` directly.

I've tried this before and didn't have any success. I still get the same error. I get the error even if I remove the gpu_id argument when calling test.py and relying on the default. It appears to be related to the the use_augment argument, which isn't recognizing 0, which make sense when I look at the argument in base_options.py:

`parser.add_argument('--use_augment', action='store_true', help='using data augmentation for training')
#parser.add_argument('--use_augment', type=int, default=1, help='using data augmentation for training')`

From my understanding of your code, this argument is used in both deepcrack_dataset.py and roadnet_dataset.py to apply affine transforms with `if self.opt.use_augment: ...`. I assume that your intent is to set this to false. How do I do this with an action argument? Can I simply remove the argument from test_roadnet.sh?

**Comment by yhlleo** [comment]
Hi @jonahduncan , 

I checked again, it's an error of the released codes. I revised the `use_augment` type from `int` to `action` (see [details here](https://github.com/yhlleo/DeepSegmentor/blob/master/options/base_options.py#L61-L62)).  So, you're right to simply delete the line `--use augment 0` from the script to keep the parameter value to false.

To test the pre-trained model, you're right to rename the file.
